File: 102_Binary_Tree_Level_Order_Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int> > result;
        queue<TreeNode*> cur;
        cur.push(root);
        TreeNode *t;
        int levelSize = root? 1 : 0;
        while (levelSize > 0){
            vector<int> thisLevel;
            int i = levelSize;
            levelSize = 0;
            for (; i > 0; i--) {
                t = cur.front();
                thisLevel.push_back(t->val);
                if (t->left) {
                    cur.push(t->left);
                    levelSize++;
                }
                if (t->right) {
                    cur.push(t->right);
                    levelSize++;
                }
                cur.pop();
            }
            result.push_back(thisLevel);
        }
        return result;
    }
};

File: 102_Binary_Tree_Level_Order_Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if root is None:
            return []
        queue, res = [root], []
        while queue:
            next, level = [], []
            for node in queue:
                level.append(node.val)
                if node.left is not None:
                    next.append(node.left)
                if node.right is not None:
                    next.append(node.right)
            res.append(level)
            queue = next
        return res
